### PR TITLE
Add validation for Deployment queue-name to fail fast

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Add validation for Deployment queue-name to fail fast

#### Which issue(s) this PR fixes:

Prepare #3528

#### Does this PR introduce a user-facing change?

```release-note
Added validation for Deployment queue-name to fail fast
```

### CODE REVIEW

**Comment by mimowo** [comment]
Adding validation is tricky. Can you please summarize if the newly added validation prevents situations which would fail later anyway? Or it could work and we now start rejecting?

**Comment by mbobrovskyi** [comment]
> Adding validation is tricky. Can you please summarize if the newly added validation prevents situations which would fail later anyway? Or it could work and we now start rejecting?

Yes, it should prevent fail later on Pods.

**Comment by mimowo** [comment]
Thanks, yeah the validation is just a fail fast - it would not work before. In that case I'm ok to consider it a bugfix and include in 0.9.2, but it feels too late for 0.9.1.

cc @tenzen-y

**Comment by tenzen-y** [comment]
> Thanks, yeah the validation is just a fail fast - it would not work before. In that case I'm ok to consider it a bugfix and include in 0.9.2, but it feels too late for 0.9.1.

Yeah, this is the change that either enhances or bugs since even if this commit does not exist, the scheduling mechanism will not be broken. But, since this would be helpful valuation, I'm ok with cherry-pick to v0.9.

As @mimowo mentioned, the 0.9.1 patch commit has already closed. So, if users want to use this validation, they can use the latest release-0.9 branch image.

File: pkg/controller/jobframework/validation.go [hunk 1]
```diff
@@ -80,26 +80,33 @@ func validateCreateForQueueName(job GenericJob) field.ErrorList {
 	return allErrs
 }
 
-func ValidateAnnotationAsCRDName(job GenericJob, crdNameAnnotation string) field.ErrorList {
+func ValidateAnnotationAsCRDName(annotations map[string]string, crdNameAnnotation string) field.ErrorList {
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply pkg/controller/jobframework/validation.go]: Done

File: pkg/controller/jobs/pod/pod_webhook.go [hunk 2]
```diff
@@ -288,7 +288,7 @@ func validatePodGroupMetadata(p *Pod) field.ErrorList {
 			))
 		}
 	} else {
-		allErrs = append(allErrs, jobframework.ValidateLabelAsCRDName(p, GroupNameLabel)...)
+		allErrs = append(allErrs, jobframework.ValidateLabelAsCRDName(p.Object().GetLabels(), GroupNameLabel)...)
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply pkg/controller/jobs/pod/pod_webhook.go]: Done

**Review by tenzen-y — COMMENTED** [review]
Thx
/lgtm
/approve

**Comment by mbobrovskyi** [comment]
Ah, we also need to cherry-pick https://github.com/kubernetes-sigs/kueue/pull/3556 , https://github.com/kubernetes-sigs/kueue/pull/3558 and https://github.com/kubernetes-sigs/kueue/pull/3557.

**Comment by mimowo** [comment]
@mbobrovskyi I checked locally that https://github.com/kubernetes-sigs/kueue/pull/3556 produces a conflict. 
Can you please prepare a PR with there 3 preparatory commits.

**Pull request merged**